File: VoiceRecorder/NoteViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using VoiceRecorder.Audio;
using GalaSoft.MvvmLight;

namespace VoiceRecorder
{
    public class NoteViewModel : ViewModelBase
    {
        public NoteViewModel(Note note, string displayName)
        {
            this.Note = note;
            this.DisplayName = displayName;
        }

        public Note Note { get; set; }
        private bool selected;
        public bool Selected
        {
            get { return selected; }
            set
            {
                if (selected != value)
                {
                    selected = value;
                    RaisePropertyChanged("Selected");
                }
            }
        }
        public string DisplayName { get; set; }
    }
}

File: ~VB Port/VoiceRecorder/NoteViewModel.vb
Imports System.Text
Imports VoiceRecorder.Audio
Imports GalaSoft.MvvmLight

Namespace VoiceRecorder
    Public Class NoteViewModel
        Inherits ViewModelBase
        Public Sub New(ByVal note As Note, ByVal displayName As String)
            Me.Note = note
            Me.DisplayName = displayName
        End Sub

        Public Property Note As Note

        Private _selected As Boolean
        Public Property Selected As Boolean
            Get
                Return _selected
            End Get
            Set(ByVal value As Boolean)
                If _selected <> value Then
                    _selected = value
                    RaisePropertyChanged("Selected")
                End If
            End Set
        End Property

        Public Property DisplayName As String
    End Class
End Namespace
